### ISSUE TITLE
Sysrandom problems

### ISSUE THREAD

**Issue by roadhazard** [Joe-Klauza/sandstorm-admin-wrapper#135]
So I ran the Windows batch file and it eventually fails with this error:

`
An error occurred while installing sysrandom (1.0.5), and Bundler cannot
continue.

In Gemfile:
  sysrandom
Failed to run Ruby Bundler. Ensure Ruby is installed with the bundler gem. https://rubyinstaller.org/downloads/
Press any key to continue . . .
`

So I try to install sysrandom manually:

`
C:\Ruby32-x64\bin>gem install sysrandom
Using rubygems directory: C:/Users/somebody/.local/share/gem/ruby/3.2.0
Temporarily enhancing PATH for MSYS/MINGW...
Building native extensions. This could take a while...
ERROR:  Error installing sysrandom:
ERROR: Failed to build gem native extension.

current directory: C:/Users/somebody/.local/share/gem/ruby/3.2.0/gems/sysrandom-1.0.5/ext/sysrandom
C:/Ruby32-x64/bin/ruby.exe extconf.rb
creating Makefile

current directory: C:/Users/somebody/.local/share/gem/ruby/3.2.0/gems/sysrandom-1.0.5/ext/sysrandom
make DESTDIR\= sitearchdir\=./.gem.20230503-23424-knmti5 sitelibdir\=./.gem.20230503-23424-knmti5 clean
current directory: C:/Users/somebody/.local/share/gem/ruby/3.2.0/gems/sysrandom-1.0.5/ext/sysrandom
make DESTDIR\= sitearchdir\=./.gem.20230503-23424-knmti5 sitelibdir\=./.gem.20230503-23424-knmti5
make failedNo such file or directory - make

Gem files will remain installed in C:/Users/somebody/.local/share/gem/ruby/3.2.0/gems/sysrandom-1.0.5 for inspection.
Results logged to C:/Users/somebody/.local/share/gem/ruby/3.2.0/extensions/x64-mingw-ucrt/3.2.0/sysrandom-1.0.5/gem_make.out
`

I went into the Ruby dir and tried:

`
gem install bundle
`

....then tried installing sysrandom again and running the Windows batch file and same problem. What am I doing wrong?

**Comment by Joe-Klauza** [comment]
Hi @roadhazard. What does the log file mentioned here show?
```
Results logged to C:/Users/somebody/.local/share/gem/ruby/3.2.0/extensions/x64-mingw-ucrt/3.2.0/sysrandom-1.0.5/gem_make.out
```
Also what specific Windows OS are you installing on?

**Comment by JckHmr89** [comment]
Currently having the same issue with this. The logfile reads:

current directory: C:/Ruby32-x64/lib/ruby/gems/3.2.0/gems/sysrandom-1.0.5/ext/sysrandom
C:/Ruby32-x64/bin/ruby.exe extconf.rb
creating Makefile

current directory: C:/Ruby32-x64/lib/ruby/gems/3.2.0/gems/sysrandom-1.0.5/ext/sysrandom
make DESTDIR\= sitearchdir\=./.gem.20230719-10468-q1z66i sitelibdir\=./.gem.20230719-10468-q1z66i clean
current directory: C:/Ruby32-x64/lib/ruby/gems/3.2.0/gems/sysrandom-1.0.5/ext/sysrandom
make DESTDIR\= sitearchdir\=./.gem.20230719-10468-q1z66i sitelibdir\=./.gem.20230719-10468-q1z66i
make failedNo such file or directory - make

**Comment by JckHmr89** [comment]
Update,
Windows 10 fresh install
Ruby 3.1.2 (non dev install)
Fresh wrapper extraction
Running the windows-install, some of the gems installed the first time and now this is the loop. Manually attempting to install from "gem install sysrandom" and other files results in a similar issue of failing to make a directory. Both the .bat and the powershell were launched as an admin, not that it made a difference

Bundling
Fetching gem metadata from https://rubygems.org/.......
Resolving dependencies...
Using bundler 2.3.7
Using ruby2_keywords 0.0.5
Using citrus 3.0.2
Using coderay 1.1.3
Using ffi 1.15.5 (x64-mingw-ucrt)
Using sync 0.5.0
Using geocoder 1.8.1
Using inifile 3.0.0
Using method_source 1.0.0
Using rack 2.2.6.4
Using rubyzip 2.3.2
Using tilt 2.0.11
Using unicode-display_width 2.3.0
Using webrick 1.7.0
Using bindata 2.4.14
Fetching oj 3.13.23
Fetching openssl 3.0.1
Fetching sysrandom 1.0.5
Fetching bcrypt 3.1.18
Using tins 1.31.1
Using mustermann 3.0.0
Using pry 0.14.1
Using rack-protection 3.0.4
Using sys-proctable 1.2.7
Using toml-rb 2.2.0
Using sinatra 3.0.4
Fetching file-tail 1.2.0
Installing sysrandom 1.0.5 with native extensions
Installing bcrypt 3.1.18 with native extensions
Installing file-tail 1.2.0
Installing openssl 3.0.1 with native extensions
Installing oj 3.13.23 with native extensions
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: C:/Ruby/lib/ruby/gems/3.1.0/gems/bcrypt-3.1.18/ext/mri
C:/Ruby/bin/ruby.exe -I C:/Ruby/lib/ruby/3.1.0 -r
./siteconf20230719-1488-gg831s.rb extconf.rb
creating Makefile

current directory: C:/Ruby/lib/ruby/gems/3.1.0/gems/bcrypt-3.1.18/ext/mri
make DESTDIR\= clean
current directory: C:/Ruby/lib/ruby/gems/3.1.0/gems/bcrypt-3.1.18/ext/mri
make DESTDIR\=
make failedNo such file or directory - make

Gem files will remain installed in
C:/Ruby/lib/ruby/gems/3.1.0/gems/bcrypt-3.1.18 for inspection.
Results logged to
C:/Ruby/lib/ruby/gems/3.1.0/extensions/x64-mingw-ucrt/3.1.0/bcrypt-3.1.18/gem_make.out

  C:/Ruby/lib/ruby/3.1.0/rubygems/ext/builder.rb:70:in `rescue in run'
  C:/Ruby/lib/ruby/3.1.0/rubygems/ext/builder.rb:67:in `run'
  C:/Ruby/lib/ruby/3.1.0/rubygems/ext/builder.rb:44:in `block in make'
  C:/Ruby/lib/ruby/3.1.0/rubygems/ext/builder.rb:36:in `each'
  C:/Ruby/lib/ruby/3.1.0/rubygems/ext/builder.rb:36:in `make'
  C:/Ruby/lib/ruby/3.1.0/rubygems/ext/ext_conf_builder.rb:63:in `block in build'
  C:/Ruby/lib/ruby/3.1.0/tempfile.rb:317:in `open'
  C:/Ruby/lib/ruby/3.1.0/rubygems/ext/ext_conf_builder.rb:26:in `build'
  C:/Ruby/lib/ruby/3.1.0/rubygems/ext/builder.rb:161:in `build_extension'
C:/Ruby/lib/ruby/3.1.0/rubygems/ext/builder.rb:195:in `block in
build_extensions'
  C:/Ruby/lib/ruby/3.1.0/rubygems/ext/builder.rb:192:in `each'
  C:/Ruby/lib/ruby/3.1.0/rubygems/ext/builder.rb:192:in `build_extensions'
  C:/Ruby/lib/ruby/3.1.0/rubygems/installer.rb:853:in `build_extensions'
C:/Ruby/lib/ruby/3.1.0/bundler/rubygems_gem_installer.rb:71:in
`build_extensions'
  C:/Ruby/lib/ruby/3.1.0/bundler/rubygems_gem_installer.rb:28:in `install'
  C:/Ruby/lib/ruby/3.1.0/bundler/source/rubygems.rb:204:in `install'
  C:/Ruby/lib/ruby/3.1.0/bundler/installer/gem_installer.rb:54:in `install'
C:/Ruby/lib/ruby/3.1.0/bundler/installer/gem_installer.rb:16:in
`install_from_spec'
C:/Ruby/lib/ruby/3.1.0/bundler/installer/parallel_installer.rb:186:in
`do_install'
C:/Ruby/lib/ruby/3.1.0/bundler/installer/parallel_installer.rb:177:in `block
in worker_pool'
  C:/Ruby/lib/ruby/3.1.0/bundler/worker.rb:62:in `apply_func'
  C:/Ruby/lib/ruby/3.1.0/bundler/worker.rb:57:in `block in process_queue'
  C:/Ruby/lib/ruby/3.1.0/bundler/worker.rb:54:in `loop'
  C:/Ruby/lib/ruby/3.1.0/bundler/worker.rb:54:in `process_queue'
C:/Ruby/lib/ruby/3.1.0/bundler/worker.rb:91:in `block (2 levels) in
create_threads'

An error occurred while installing bcrypt (3.1.18), and Bundler cannot continue.

In Gemfile:
  bcrypt


Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

current directory:
C:/Ruby/lib/ruby/gems/3.1.0/gems/sysrandom-1.0.5/ext/sysrandom
C:/Ruby/bin/ruby.exe -I C:/Ruby/lib/ruby/3.1.0 -r
./siteconf20230719-1488-ooy1on.rb extconf.rb
creating Makefile

current directory:
C:/Ruby/lib/ruby/gems/3.1.0/gems/sysrandom-1.0.5/ext/sysrandom
make DESTDIR\= clean
current directory:
C:/Ruby/lib/ruby/gems/3.1.0/gems/sysrandom-1.0.5/ext/sysrandom
make DESTDIR\=
make failedNo such file or directory - make

Gem files will remain installed in
C:/Ruby/lib/ruby/gems/3.1.0/gems/sysrandom-1.0.5 for inspection.
Results logged to
C:/Ruby/lib/ruby/gems/3.1.0/extensions/x64-mingw-ucrt/3.1.0/sysrandom-1.0.5/gem_make.out

  C:/Ruby/lib/ruby/3.1.0/rubygems/ext/builder.rb:70:in `rescue in run'
  C:/Ruby/lib/ruby/3.1.0/rubygems/ext/builder.rb:67:in `run'
  C:/Ruby/lib/ruby/3.1.0/rubygems/ext/builder.rb:44:in `block in make'
  C:/Ruby/lib/ruby/3.1.0/rubygems/ext/builder.rb:36:in `each'
  C:/Ruby/lib/ruby/3.1.0/rubygems/ext/builder.rb:36:in `make'
  C:/Ruby/lib/ruby/3.1.0/rubygems/ext/ext_conf_builder.rb:63:in `block in build'
  C:/Ruby/lib/ruby/3.1.0/tempfile.rb:317:in `open'
  C:/Ruby/lib/ruby/3.1.0/rubygems/ext/ext_conf_builder.rb:26:in `build'
  C:/Ruby/lib/ruby/3.1.0/rubygems/ext/builder.rb:161:in `build_extension'
C:/Ruby/lib/ruby/3.1.0/rubygems/ext/builder.rb:195:in `block in
build_extensions'
  C:/Ruby/lib/ruby/3.1.0/rubygems/ext/builder.rb:192:in `each'
  C:/Ruby/lib/ruby/3.1.0/rubygems/ext/builder.rb:192:in `build_extensions'
  C:/Ruby/lib/ruby/3.1.0/rubygems/installer.rb:853:in `build_extensions'
C:/Ruby/lib/ruby/3.1.0/bundler/rubygems_gem_installer.rb:71:in
`build_extensions'
  C:/Ruby/lib/ruby/3.1.0/bundler/rubygems_gem_installer.rb:28:in `install'
  C:/Ruby/lib/ruby/3.1.0/bundler/source/rubygems.rb:204:in `install'
  C:/Ruby/lib/ruby/3.1.0/bundler/installer/gem_installer.rb:54:in `install'
C:/Ruby/lib/ruby/3.1.0/bundler/installer/gem_installer.rb:16:in
`install_from_spec'
C:/Ruby/lib/ruby/3.1.0/bundler/installer/parallel_installer.rb:186:in
`do_install'
C:/Ruby/lib/ruby/3.1.0/bundler/installer/parallel_installer.rb:177:in `block
in worker_pool'
  C:/Ruby/lib/ruby/3.1.0/bundler/worker.rb:62:in `apply_func'
  C:/Ruby/lib/ruby/3.1.0/bundler/worker.rb:57:in `block in process_queue'
  C:/Ruby/lib/ruby/3.1.0/bundler/worker.rb:54:in `loop'
  C:/Ruby/lib/ruby/3.1.0/bundler/worker.rb:54:in `process_queue'
C:/Ruby/lib/ruby/3.1.0/bundler/worker.rb:91:in `block (2 levels) in
create_threads'

An error occurred while installing sysrandom (1.0.5), and Bundler cannot
continue.

In Gemfile:
  sysrandom
Failed to run Ruby Bundler. Ensure Ruby is installed with the bundler gem. https://rubyinstaller.org/downloads/
Press any key to continue . . .

**Comment by JckHmr89** [comment]
I fixed the issue, I installed Ruby 3.1.2 Dev-kit instead.

**Comment by Joe-Klauza** [comment]
Thanks for the info and solution @JckHmr89. I have added info to the `README.md` to explicitly mention the Devkit. I'm going to close this issue, but @roadhazard feel free to reopen this or open a new issue if use of the Devkit installation does not correct the error for you.
